src/proposal.mligo

```diff
@@ -146,6 +170,14 @@ let freeze_on_ledger (tokens, addr, ledger, total_supply, frozen_token_id, gover
 
 let stake_tk(token_amount, addr, voting_period, store : nat * address * voting_period * storage): storage =
   let current_period = get_current_period_num(store.start_time, voting_period) in
+  let new_period_staked = match is_nat(current_period - store.period_staked.last_updated_period) with
+    | Some elapsed -> if elapsed >= 2n //TODO@SRAS Add test to check staked tokens in a proposing period and voting period after it are both accounted towards the period no of the proposing period
```

Comment:
As I was suggesting, storing the staked total in the `quorum_threshold_at_stage` might make this easier (not the test, but the logic).

We'll need to call `update_quorum` in the beginning of the `propose` and `vote` entrypoint, but then we can just add to the staked total after that.

Reply:
>and vote entrypoint,

Is calling `update_qourum` in `vote` required? given than `vote` cannot happen without a `propose` operation that preceds it, and we update the quorum for the whole cycle there?

Reply:
Just repeating: I agree that `update_quorum` is not needed in `vote`, but I still think we can store the total staked alongside the quorum.

In which IIUC we still don't need to check the period here in `stake_tk` and we can add to the `total_staked_cycle` that I'm proposing.

Reply:
Have updated.